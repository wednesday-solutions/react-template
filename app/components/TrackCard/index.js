/**
 *
 * TrackCard
 *
 */

import React from 'react';
import { Card } from 'antd';
import styled from 'styled-components';
import { PlayCircleFilled, PauseCircleFilled } from '@ant-design/icons';
import If from '@components/If';
import { PropTypes } from 'prop-types';
import { injectIntl } from 'react-intl';
const { Meta } = Card;

const CustomCard = styled(Card)`
  && {
    margin: 20px 0;
    max-height: 30rem;
    ${props => props.color && `color: ${props.color}`};
  }
`;
const MetaCard = styled(Meta)`
  && {
    margin: 0.5rem !important;
  }
`;
const TrackCardContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
const CustomPlay = styled(PlayCircleFilled)`
  font-size: 2rem;
  padding: 1rem;
  cursor: pointer;
`;
const CustomPause = styled(PauseCircleFilled)`
  font-size: 2rem;
  padding: 1rem;
  cursor: pointer;
`;
const ProgressBar = styled.progress`
  && {
    display: block;
    font-size: 1rem;
  }
`;

function TrackCard({ track, currentlyPlayingTrackId, currentTrack, onPlay, onPause, isPlaying }) {
  return (
    <CustomCard data-testid="track-card" cover={<img alt="artwork-url" src={track.artworkUrl100} />}>
      <TrackCardContainer>
        <If condition={isPlaying && track.trackId === currentlyPlayingTrackId}>
          <CustomPause onClick={() => onPause(track.previewUrl)} />
        </If>
        <If condition={!(isPlaying && track.trackId === currentlyPlayingTrackId)}>
          <CustomPlay onClick={() => onPlay(track.previewUrl, track.trackId)} />
        </If>
      </TrackCardContainer>
      <ProgressBar
        value={
          track.trackId === currentlyPlayingTrackId && currentTrack.currentTime !== 0 ? currentTrack?.currentTime : 0
        }
        max={track.trackTimeMillis / 10000}
      ></ProgressBar>
      <MetaCard title={track.trackName} description={track.artistName}>
        {' '}
      </MetaCard>
    </CustomCard>
  );
}

TrackCard.propTypes = {
  onPause: PropTypes.func,
  onPlay: PropTypes.func,
  track: PropTypes.object
};

export default injectIntl(TrackCard);
