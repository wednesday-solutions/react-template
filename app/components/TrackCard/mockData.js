const MOCK_TRACK = {
  wrapperType: 'track',
  kind: 'song',
  artistId: 1447341718,
  collectionId: 1462857259,
  trackId: 1462857265,
  artistName: 'RIZE',
  collectionName: 'Journey (feat. Y Des) - Single',
  trackName: 'Journey (feat. Y Des)',
  collectionCensoredName: 'Journey (feat. Y Des) - Single',
  trackCensoredName: 'Journey (feat. Y Des)',
  artistViewUrl: 'https://music.apple.com/us/artist/rize/1447341718?uo=4',
  collectionViewUrl: 'https://music.apple.com/us/album/journey-feat-y-des/1462857259?i=1462857265&uo=4',
  trackViewUrl: 'https://music.apple.com/us/album/journey-feat-y-des/1462857259?i=1462857265&uo=4',
  previewUrl:
    'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview123/v4/33/06/2a/33062a5c-cb7f-3783-b263-295d8bebb041/mzaf_1974938250219788592.plus.aac.p.m4a',
  artworkUrl30:
    'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/88/3f/c8/883fc8f7-9174-5a86-1616-92fdd9030702/source/30x30bb.jpg',
  artworkUrl60:
    'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/88/3f/c8/883fc8f7-9174-5a86-1616-92fdd9030702/source/60x60bb.jpg',
  artworkUrl100:
    'https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/88/3f/c8/883fc8f7-9174-5a86-1616-92fdd9030702/source/100x100bb.jpg',
  collectionPrice: 0.99,
  trackPrice: 0.99,
  releaseDate: '2019-05-10T12:00:00Z',
  collectionExplicitness: 'explicit',
  trackExplicitness: 'explicit',
  discCount: 1,
  discNumber: 1,
  trackCount: 1,
  trackNumber: 1,
  trackTimeMillis: 206538,
  country: 'USA',
  currency: 'USD',
  primaryGenreName: 'Hip-Hop/Rap',
  contentAdvisoryRating: 'Explicit',
  isStreamable: true
};

export default MOCK_TRACK;
