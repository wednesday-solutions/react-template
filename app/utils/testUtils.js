import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
