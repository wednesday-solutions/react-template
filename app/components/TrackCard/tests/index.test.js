/**
 *
 * Tests for TrackCard
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import TrackCard from '../index';
import MOCK_TRACK from '../mockData';
describe('<TrackCard />', () => {
  it('should render and match the snapshot', () => {
    console.log('before testing');
    const { getAllByTestId, baseElement } = renderWithIntl(<TrackCard track={MOCK_TRACK} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 TrackCard component', () => {
    const { getAllByTestId } = renderWithIntl(<TrackCard track={MOCK_TRACK} />);
    expect(getAllByTestId('track-card').length).toBe(1);
  });
});
