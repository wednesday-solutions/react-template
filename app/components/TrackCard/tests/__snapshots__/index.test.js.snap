// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered TrackCard__CustomCard-sc-10bk196-0 krtmBA"
      data-testid="track-card"
    >
      <div
        class="ant-card-cover"
      >
        <img
          alt="artwork-url"
          src="https://is5-ssl.mzstatic.com/image/thumb/Music123/v4/88/3f/c8/883fc8f7-9174-5a86-1616-92fdd9030702/source/100x100bb.jpg"
        />
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="TrackCard__TrackCardContainer-sc-10bk196-2 jILkkF"
        >
          <span
            aria-label="play-circle"
            class="anticon anticon-play-circle TrackCard__CustomPlay-sc-10bk196-3 ldNKvG"
            role="img"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
              />
            </svg>
          </span>
        </div>
        <progress
          class="TrackCard__ProgressBar-sc-10bk196-5 eksdvV"
          max="20.6538"
          value="0"
        />
        <div
          class="ant-card-meta TrackCard__MetaCard-sc-10bk196-1 pXxsN"
        >
          <div
            class="ant-card-meta-detail"
          >
            <div
              class="ant-card-meta-title"
            >
              Journey (feat. Y Des)
            </div>
            <div
              class="ant-card-meta-description"
            >
              RIZE
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
